descartes_light/test/src/perf_plotter.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python
+
+import math
+import numpy as np
+# import matplotlib as mpl
+import matplotlib.pyplot as plt
+from mpl_toolkits.mplot3d import axes3d, Axes3D #<-- Note the capitalization! 
+import sys
+
+class Perf_Plotter():
+    def __init__(self):
+        self.dict = {}
+
+    def strip(self, filename = "../benchmarks/benchmarks.txt"):
+        #Each title in "dict" refers to a different algorythm: dict["alg_name"]
+        #Each algorithm has a dictionary of lists: dict["alg_name"]['list_name']
+        #There are five lists per alg: "dof", "n_waypoints", "samples", "Graph Build Time", "Graph Search Time"
+        #Ergo to get each data value, use "dict['alg_name']['key_name'][test#]"
+        currentGraphTitle = ' '
+        with open(filename) as input:
+            for line in input:
+                if "descartes_light::" in line:
+                    #title lines detected.
+                    currentGraphTitle = line[line.find('::')+2:-1]
+                    # self.dict[currentGraphTitle] = [] #initialize the value at this key to an empty list
+                    self.dict[currentGraphTitle] = {'dofs': [], 'n_waypoints': [], 'samples': [], 'Graph Build Time': [], 'Graph Search Time': []}
+                    print("Now reading " + currentGraphTitle)
+
+                elif 'Inputs' in line:
+                    #New datapoint for a new simulation: Added as the last dict element for this graph
+                    currentDatapointDict = self.stripInputs(line)
+                    #add a dictionary to the list of datapoint dictionaries:
+                    self.dict[currentGraphTitle]['dofs'].append(float(currentDatapointDict['dofs']))
+                    self.dict[currentGraphTitle]['n_waypoints'].append(float(currentDatapointDict['n_waypoints']))
+                    self.dict[currentGraphTitle]['samples'].append(float(currentDatapointDict['samples']))
+                    
+
+                elif 'Graph' in line:
+                    title, value = self.stripTime(line)
+                    #Add an entry to the last datapoint dict for this time value:
+                    self.dict[currentGraphTitle][title].append(float(value))
+
+
+    def stripInputs(self, textline):
```

Comment:
The convention for Python functions is `snake_case`:

```suggestion
    def strip_inputs(self, textline):
```